### PR TITLE
Repetitive log when query account from cache

### PR DESCRIPTION
**Pull request by hieunguyenmsft**
## Proposed changes

Report in this [link](https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1449130)
These messages are repetitive written:
`2021-06-25T06:33:56.781Z kWarning [MSAL:0001] WARNING -[MSAIMSIDAccountCacheItem initWithJSONDictionary:error:]:284 No account type present in the JSON for credential`

`2021-06-25T06:33:56.781Z kVerbose [MSAL:0001] DEBUG   -[MSAIMSIDCacheItemJsonSerializer deserializeCacheItem:ofClass:]:96 TID=515301 (main thread) MSAL.xplat.iOS 1.0.0+bdeeb209 iOS 14.6 [2021-06-25 06:33:56] Failed to deserialize object (null) of expected class MSAIMSIDAccountCacheItem`

`2021-06-25T06:33:56.782Z kInfo [MSAL:0001] INFO   -[MSAIMSIDKeychainTokenCache cacheItemsWithKey:serializer:cacheItemClass:context:error:]:755 TID=515301 (main thread) MSAL.xplat.iOS 1.0.0+bdeeb209 iOS 14.6 [2021-06-25 06:33:56] Failed to deserialize item with class MSAIMSIDAccountCacheItem.`

I suspect the error is because kSecAttrType is nil in the query critirical.

## Type of change

- [ ] Feature work
- [x] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/cache/key/MSIDDefaultAccountCacheQuery.m [hunk 1]
```diff
@@ -35,16 +35,6 @@ - (NSString *)account
     return nil;
```
Comment (kaisong1990) [inline IdentityCore/src/cache/key/MSIDDefaultAccountCacheQuery.m:35]: This pull request does not update changelog.txt.

Please consider if this change would be noticeable to a partner or user and either update changelog.txt or resolve this conversation.

File: IdentityCore/src/cache/key/MSIDDefaultAccountCacheQuery.m [hunk 2]
```diff
@@ -35,16 +35,6 @@ - (NSString *)account
...
 }
 
-- (NSNumber *)type
-{
-    if (self.accountType != MSIDAccountTypeOther)
```
Comment (oldalton) [inline IdentityCore/src/cache/key/MSIDDefaultAccountCacheQuery.m:40]: Would it be safer to instead default to MSSTS? I'm concerned that we will have account query somewhere in our code that doesn't set this and will fail the query.

**Review by oldalton — CHANGES_REQUESTED** [review]

File: IdentityCore/src/cache/key/MSIDDefaultAccountCacheQuery.m (hunk 2 above)
Comment (hieunguyenmsft) [reply IdentityCore/src/cache/key/MSIDDefaultAccountCacheQuery.m:40]: Sure, although I cross-checked in MSAL and broker but I might missed other libs. I will update to your suggestion.

Comment (hieunguyenmsft) [reply IdentityCore/src/cache/key/MSIDDefaultAccountCacheQuery.m:40]: Addressed

**Review by oldalton — APPROVED** [review]

**Pull request merged**